compiler/wasm/src/compile_new.rs

```diff
@@ -0,0 +1,251 @@
+    pub fn process_root_crate(&mut self, path_to_crate: String) -> CrateIDWrapper {
+        let path_to_crate = Path::new(&path_to_crate);
+
+        // Adds the root crate to the crate graph and returns its crate id
+        CrateIDWrapper(prepare_crate(&mut self.context, path_to_crate))
+    }
+
+    pub fn process_dependency_crate(&mut self, path_to_crate: String) -> CrateIDWrapper {
+        let path_to_crate = Path::new(&path_to_crate);
+
+        // Adds the root crate to the crate graph and returns its crate id
+        CrateIDWrapper(prepare_dependency(&mut self.context, path_to_crate))
+    }
+
+    // Adds a named edge from one crate to the other.
+    //
+    // For example, lets say we have two crates CrateId1 and CrateId2
+    // This function will add an edge from CrateId1 to CrateId2 and the edge will be named `crate_name`
+    //
+    // This essentially says that CrateId1 depends on CrateId2 and the dependency is named `crate_name`
+    pub fn add_dependency_edge(
+        &mut self,
+        crate_name: String,
+        from: CrateIDWrapper,
+        to: CrateIDWrapper,
+    ) {
+        let parsed_crate_name: CrateName = crate_name
+            .parse()
+            .unwrap_or_else(|_| panic!("Failed to parse crate name {}", crate_name));
+        add_dep(&mut self.context, from.0, to.0, parsed_crate_name);
+    }
+
+    pub fn compile_program(mut self) -> Result<JsCompileResult, JsCompileError> {
+        let compile_options = CompileOptions::default();
+        // For now we default to plonk width = 3, though we can add it as a parameter
+        let np_language = acvm::Language::PLONKCSat { width: 3 };
+        #[allow(deprecated)]
+        let is_opcode_supported = acvm::pwg::default_is_opcode_supported(np_language);
+
+        let root_crate_id = *self.context.root_crate_id();
+
+        let compiled_program =
+            compile_main(&mut self.context, root_crate_id, &compile_options, None, true)
+                .map_err(|errs| {
+                    CompileError::with_file_diagnostics(
+                        "Failed to compile program",
+                        errs,
+                        &self.context.file_manager,
+                    )
+                })?
+                .0;
+
+        let optimized_program =
+            nargo::ops::optimize_program(compiled_program, np_language, &is_opcode_supported)
+                .expect("Program optimization failed");
+
+        let compile_output = preprocess_program(optimized_program);
+        Ok(JsCompileResult::new(compile_output))
+    }
+
+    pub fn compile_contract(mut self) -> Result<JsCompileResult, JsCompileError> {
+        let compile_options = CompileOptions::default();
+        // For now we default to plonk width = 3, though we can add it as a parameter
+        let np_language = acvm::Language::PLONKCSat { width: 3 };
+        #[allow(deprecated)]
+        let is_opcode_supported = acvm::pwg::default_is_opcode_supported(np_language);
```

Comment:
Can we pass this in through `new()`

Reply:
Yep, we can pass in the program_width -- we could also pass in the set of supported opcodes though if we go with https://github.com/noir-lang/noir/issues/3772 then the opcode list becomes irrelevant